src/common/common.module.ts

```diff
@@ -1,8 +1,24 @@
 import { Module } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
 import { MailService } from "./mail.service";
+import { RabbitMQService } from "./rabbitmq/rabbitmq.service";
 
 @Module({
-  providers: [MailService],
-  exports: [MailService],
+  providers: [
+    MailService,
+    {
+      provide: RabbitMQService,
+      useFactory: (configService: ConfigService) => {
+        const isEnabled =
+          configService.get<string>("rabbitMq.enabled") === "yes";
+        if (isEnabled) {
+          return new RabbitMQService(configService);
+        }
+        return null;
+      },
+      inject: [ConfigService],
+    },
+  ],
```

Comment:
Wouldn't it be better to import `RabbitMQmodule` here instead of setting it as a provider? Considering rabbitmq module is created but not used anywhere

Reply:
CommonModule is a bit of a weird module. It is needed for services which are configured globally in app.module but which need to be imported by other modules.

That said, I don't really get why RabbitMQService needs to be created here rather than in its own module. @despadam do you remember the reason?